src/main/java/com/vlkan/pubsub/PubsubPublisher.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Volkan Yazıcı
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permits and
+ * limitations under the License.
+ */
+
+package com.vlkan.pubsub;
+
+import com.vlkan.pubsub.model.PubsubPublishRequest;
+import com.vlkan.pubsub.model.PubsubPublishResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+public class PubsubPublisher {
```

Comment:
Although I understand that Pubsub stands for "Publisher+Subscriber", I would avoid using `*Publisher`/`*Subscriber` naming, especially in a reactive library, unless they implement `Publisher`/`Subscriber`.

It confuses a lot, and even got me when I started the review 😅

Reply:
Agreed, but they serve another purpose here: They comply with the WYSIWYG contract of the project. There is no *nack* operator in the project, because Pub/Sub doesn't have a concept of *nack* -- ironically the official Java driver has a `nack()` method to cancel the covert lease time extending background task. When we say *publish*, it is a Pub/Sub *publish*; when we say *pull*, it is a Pub/Sub *pull*, and so on.

But as I admitted, your complaint is not groundless. Do you have any suggestions?

Reply:
PubSub{Sender,Receiver} could work. Semantically the same, but does not 1:1 match to the well-known RS Publisher/Subscriber

Reply:
`Pubsub{Sender,Receiver}` sounds good enough. One more question: Shall I rename the rest of the entities and methods accordingly? That is,

- `PubsubPublishRequest` :arrow_right: `PubsubSendRequest`
- `PubsubPublishResponse` :arrow_right: `PubsubSendResponse`
- `PubsubPublisher#publish` :arrow_right: `PubsubSender#send`
- `PubsubSubscriber#pull` :arrow_right: `PubsubReceiver#receive`
- `PubsubPullRequest` :arrow_right: `PubsubReceiveRequest`
- `PubsubPullResponse` :arrow_right: `PubsubReceiveResponse`

I am not 100% convinced about this change since it is sort of violating the WYSIWYG contract. OTOH, I can understand this smells really bad to an RS library maintainer. I am sort of sitting on the fence right now.

Reply:
`*Publish{Request,Response}` sounds okay, as well as `publish/pull` methods. Just make sure they are not `subscribe/onNext` :D